Editor/Utility/SaveDuringPlay.cs

```diff
@@ -165,6 +166,43 @@ bool ScanFields(string fullName, Type type, ref object obj)
                     if (doneSomething)
                         obj = array;
                 }
+                else if (typeof(IList).IsAssignableFrom(type))
+                {
+                    isLeaf = false;
+                    var list = obj as IList;
+                    object length = list.Count;
+                    
+                    // restore list size
+                    if (OnLeafField != null && OnLeafField(
+                        fullName + ".Length", length.GetType(), ref length))
+                    {
+                        var newLength = (int)length;
+                        var currentLength = list.Count;
+                        for (int i = 0; i < currentLength - newLength; ++i)
+                        {
+                            list.RemoveAt(0); // make list shorter if needed
```

Comment:
Just change this to `list.RemoveAt(currentLength - i - 1);` to avoid the n^2 memcopies

Reply:
Good point 👍